ProcessLib/ThermoHydroMechanics/CreateThermoHydroMechanicsProcess.cpp

```diff
@@ -0,0 +1,310 @@
+
+    // Process variable.
+
+    //! \ogs_file_param{prj__processes__process__THERMO_HYDRO_MECHANICS__process_variables}
+    auto const pv_config = config.getConfigSubtree("process_variables");
+
+    ProcessVariable* variable_T;
+    ProcessVariable* variable_p;
+    ProcessVariable* variable_u;
+    std::vector<std::vector<std::reference_wrapper<ProcessVariable>>>
+        process_variables;
+    if (use_monolithic_scheme)  // monolithic scheme.
+    {
+        auto per_process_variables = findProcessVariables(
+            variables, pv_config,
+            {//! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS__process_variables__temperature}
+             "temperature",
+             //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS__process_variables__pressure}
+             "pressure",
+             //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS__process_variables__displacement}
+             "displacement"});
+        variable_T = &per_process_variables[0].get();
+        variable_p = &per_process_variables[1].get();
+        variable_u = &per_process_variables[2].get();
+        process_variables.push_back(std::move(per_process_variables));
+    }
+    else  // staggered scheme.
+    {
+        using namespace std::string_literals;
+        for (auto const& variable_name :
+             {"Temperature"s, "pressure"s, "displacement"s})
+        {
+            auto per_process_variables =
+                findProcessVariables(variables, pv_config, {variable_name});
+            process_variables.push_back(std::move(per_process_variables));
+        }
+        variable_T = &process_variables[0][0].get();
+        variable_p = &process_variables[1][0].get();
+        variable_u = &process_variables[2][0].get();
+    }
+
+    DBUG("Associate displacement with process variable \'%s\'.",
+         variable_u->getName().c_str());
+
+    if (variable_u->getNumberOfComponents() != DisplacementDim)
+    {
+        OGS_FATAL(
+            "Number of components of the process variable '%s' is different "
+            "from the displacement dimension: got %d, expected %d",
+            variable_u->getName().c_str(),
+            variable_u->getNumberOfComponents(),
+            DisplacementDim);
+    }
+
+    DBUG("Associate pressure with process variable \'%s\'.",
+         variable_p->getName().c_str());
+    if (variable_p->getNumberOfComponents() != 1)
+    {
+        OGS_FATAL(
+            "Pressure process variable '%s' is not a scalar variable but has "
+            "%d components.",
+            variable_p->getName().c_str(),
+            variable_p->getNumberOfComponents());
+    }
+
+    DBUG("Associate temperature with process variable \'%s\'.",
+         variable_T->getName().c_str());
+    if (variable_T->getNumberOfComponents() != 1)
+    {
+        OGS_FATAL(
+            "temperature process variable '%s' is not a scalar variable but "
+            "has "
```

Comment:
Please merge this line and the line below.

Reply:
Corrected

Reply:
Not yet.

Reply:
Corrected this time. Sorry for the mistake before... Weird

Reply:
This is not corrected. Please upload the changes first, then if and only if the change is appearing in the diff-view, mark it as corrected.

Reply:
Still not fixed!